client/src/components/graphs/Pie.js

```diff
@@ -1,50 +1,38 @@
+import React, { useRef, useEffect } from "react"
+import PropTypes from "prop-types"
 import SVGCanvas from "components/graphs/SVGCanvas"
 import * as d3 from "d3"
-import PropTypes from "prop-types"
 
-export default class Pie extends SVGCanvas {
-  static propTypes = {
-    label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    segmentFill: PropTypes.func.isRequired,
-    segmentLabel: PropTypes.func.isRequired,
-    data: PropTypes.object.isRequired
-  }
+const Pie = props => {
```

Comment:
Prefer `const Pie = ({ label, …etc… }) => {` instead of referring to `props.something` everywhere.